pkg/machines/components/delete.jsx

```diff
@@ -0,0 +1,123 @@
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Modal, OverlayTrigger, Tooltip } from 'patternfly-react';
+
+import cockpit from 'cockpit';
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+
+const _ = cockpit.gettext;
+
+export class Delete extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showModal: false,
+            dialogError: undefined,
+        };
+
+        this.delete = this.delete.bind(this);
+        this.close = this.close.bind(this);
+        this.open = this.open.bind(this);
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+    }
+
+    delete() {
+        this.props.deleteHandler()
+                .fail(exc => this.dialogErrorSet(cockpit.format(_("The $0 could not be deleted"), this.props.objectType.toLowerCase()), exc.message))
+                .then(exc => this.close());
+    }
+
+    open() {
+        this.setState({ showModal: true });
+    }
+
+    close() {
+        this.setState({ showModal: false, dialogError: undefined });
+    }
+
+    dialogErrorSet(text, detail) {
+        this.setState({ dialogError: text, dialogErrorDetail: detail });
+    }
+
+    render() {
+        const { objectName, objectType, objectId, disabled, overlayText, actionName } = this.props;
+
+        const deleteButton = () => {
+            if (disabled) {
+                return (
+                    <OverlayTrigger overlay={
+                        <Tooltip id={`delete-${objectId}-tooltip`}>
+                            { overlayText }
+                        </Tooltip> } placement='top'>
+                        <span>
+                            <Button id={`delete-${objectId}`}
+                                bsStyle='danger'
+                                style={{ pointerEvents: 'none' }}
+                                disabled>
+                                {actionName || _("Delete")}
+                            </Button>
+                        </span>
+                    </OverlayTrigger>
+                );
+            } else {
+                return (
+                    <Button id={`delete-${objectId}`}
+                        bsStyle='danger'
+                        onClick={this.open}>
+                        {actionName || _("Delete")}
+                    </Button>
+                );
+            }
+        };
+
+        return (
+            <React.Fragment>
+                { deleteButton() }
+
+                <Modal show={this.state.showModal} onHide={this.close}>
+                    <Modal.Header>
+                        <Modal.CloseButton onClick={this.close} />
+                        <Modal.Title>{ (actionName || _("Delete")) + cockpit.format((" $0 $1"), objectType, objectName) }</Modal.Title>
+                    </Modal.Header>
+                    <Modal.Body>
+                        { cockpit.format(_("Confirm this action")) }
```

Comment:
This looks weird. I think it need design approve from @andreasn or @garrett

Reply:
Alternativelly:
- it could say "Confirm this deletion/removal/detachment"
- there would be nothing in modal body, and instead the title would say "Confirm Deletion of Network Device 52:54:00:83:a8:83"
- In the dialog, instead of button which now says "Delete/Remove/Detach...", that button could say "Confirm"